libpromises/evalfunction.c

```diff
@@ -7153,7 +7153,8 @@ static FnCallResult FnCallReadJson(ARG_UNUSED EvalContext *ctx,
 
 static FnCallResult ValidateDataGeneric(const char *const fname,
                                         const char *data,
-                                        const DataFileType requested_mode)
+                                        const DataFileType requested_mode,
+                                        bool nojsonprimitive)
```

Comment:
I don't think `nojsonprimitive` is a good name - in general thinking in negatives can be tricky, and in this case it seems ambiguous. I'd go with a simple `strict` to make the validation more strict, and the details can be found in the parameter description and comments;

 ```suggestion
                                         bool strict)
```